### ISSUE TITLE
Unable to mount docker-socket with v0.4.0

### ISSUE THREAD

**Issue by badeball** [nestybox/sysbox#404]
Due to legacy reasons, we're still mounting the host's docker-socket into our containers. With v0.3.0 and by using `setfacl` on the socket, this worked fine. With v0.4.0 however, this seems to have ceased to work. I've put together an [example](https://gist.github.com/badeball/9ff3bd8caa2bdc0714e1274a54b01f36) using Vagrant to provision a VM that illustrates inability to call eg. `docker info` in such container.

There's nothing very particular going on in the Vagrantfile, `prepare.sh` or `verify.sh`. I pretty much just install sysbox, configure Docker and invoke `setfacl` with the IDs that sysbox has determined.

You can run the following to watch it fail on v0.4.0:

```
$ git clone https://gist.github.com/9ff3bd8caa2bdc0714e1274a54b01f36.git
$ cd 9ff3bd8caa2bdc0714e1274a54b01f36
$ vagrant up
...
    default: ++ docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock docker:latest docker info
    default: Client:
    default:  Context:    default
    default:  Debug Mode: false
    default: 
    default: Server:
    default: ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```
.. and run the following to watch it succeed on v0.3.0:

```
$ git checkout master~
$ vagrant destroy -f
$ vagrant up
...
    default: ++ docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock docker:latest docker info
    default: WARNING: No swap limit support
    default: Client:
    default:  Context:    default
    default:  Debug Mode: false
    default: 
    default: Server:
    default:  Containers: 1
    default:   Running: 1
    default:   Paused: 0
    default:   Stopped: 0
    default:  Images: 1
    default:  Server Version: 20.10.7
    default:  Storage Driver: overlay2
    default:   Backing Filesystem: extfs
    default:   Supports d_type: true
    default:   Native Overlay Diff: true
    default:   userxattr: false
    default:  Logging Driver: json-file
    default:  Cgroup Driver: cgroupfs
    default:  Cgroup Version: 1
    default:  Plugins:
    default:   Volume: local
    default:   Network: bridge host ipvlan macvlan null overlay
    default:   Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
    default:  Swarm: inactive
    default:  Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc sysbox-runc
    default:  Default Runtime: sysbox-runc
    default:  Init Binary: docker-init
    default:  containerd version:
    default:  runc version:
    default:  init version:
    default:  Security Options:
    default:   apparmor
    default:   seccomp
    default:    Profile: default
    default:  Kernel Version: 5.4.0-88-generic
    default:  Operating System: Ubuntu 20.04.3 LTS
    default:  OSType: linux
    default:  Architecture: x86_64
    default:  CPUs: 2
    default:  Total Memory: 981MiB
    default:  Name: ubuntu-focal
    default:  ID: ZZ2B:ZM27:SGSN:7OVN:XDWS:5UJC:C6SI:Z6ZW:LECS:HOB4:KUHH:SINK
    default:  Docker Root Dir: /var/lib/docker
    default:  Debug Mode: false
    default:  Registry: https://index.docker.io/v1/
    default:  Labels:
    default:  Experimental: false
    default:  Insecure Registries:
    default:   127.0.0.0/8
    default:  Live Restore Enabled: false
    default: 
```

Run to clean up:

```
$ vagrant destroy -f
$ cd ..
$ rm -rf 9ff3bd8caa2bdc0714e1274a54b01f36
```

**Comment by badeball** [comment]
If you're unfamiliar with Vagrant, you can run `vagrant ssh` (in the directory where you ran `vagrant up`) to SSH into the VM, where you have sudo and can mess around.

**Comment by ctalledo** [comment]
FYI, this will be fixed in the upcoming Sysbox v0.5.0 release by virtue of Sysbox supporting the Linux "ID-mapped mounts" feature (alternative to shiftfs that works on socket mounts). However, this requires a host with kernel >= 5.12.

**Comment by ctalledo** [comment]
Closing as this is fixed in the upstream code, and fix will be present in upcoming v0.5.0 release.
